src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/RemoveEmptyFinalizers.cs

```diff
@@ -33,31 +35,45 @@ public override void Initialize(AnalysisContext analysisContext)
             analysisContext.EnableConcurrentExecution();
             analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            analysisContext.RegisterCodeBlockAction(codeBlockContext =>
+            analysisContext.RegisterOperationBlockStartAction(obsac =>
             {
-                if (codeBlockContext.OwningSymbol.Kind != SymbolKind.Method)
+                if (!obsac.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemDiagnosticsConditionalAttribute, out var conditionalAttributeType) ||
+                    obsac.OperationBlocks.Length != 1 ||
+                    !(obsac.OperationBlocks[0] is IBlockOperation blockOperation) ||
+                    !(obsac.OwningSymbol is IMethodSymbol methodSymbol) ||
```

Comment:
Is there any real benefit for me to check the `Kind` before doing the cast or is the performance gain negligible? (e.g. instead of `!(obsac.OwningSymbol is IMethodSymbol methodSymbol)` I could write `obsac.OwningSymbol.Kind != SymbolKind.MethodSymbol` and do the cast afterward).